### PR TITLE
chore: bundle sqlite by default for tests

### PR DESCRIPTION
**Pull request by stringhandler**
Description
---
Yet another attempt at bundling openssl and sqlite for easy debugging/testing on Windows

Motivation and Context
---
The current feature setup means that you cannot run `cargo test` in the `wallet_ffi` or any specific folder other than the root. I tried to follow the same way that other crates do this, but allowing a user to optionally specify whether a dependency is bundled or not, and not bundle it by default, but this makes running individual tests difficult. 

Since tests are run regularly, I made the default on wallet_ffi include the bundles, but then added `--no-default-features` to the mobile builds 

How Has This Been Tested?
---
Cargo test

### CODE REVIEW

**Comment by delta1** [comment]
LGTM just waiting on CI

**Review by delta1 — DISMISSED** [review]

**Review by philipr-za — DISMISSED** [review]
LGTM

File: base_layer/wallet_ffi/mobile_build.sh
Comment (StriderDM) [inline base_layer/wallet_ffi/mobile_build.sh]: Are we only keeping support for the latest macOS? If not, this will need to be updated as well.

Comment (stringhandler) [reply base_layer/wallet_ffi/mobile_build.sh]: didn't realise there were so many logic branches here

**Comment by StriderDM** [comment]
Looks like it could work, although I think we should do this before merging this in:
mobile_build.sh has to run and produce libraries for both android and iOS.
The wallet projects have to be built and run with these libraries (disable automatic download of dependencies or they will be overwritten) on the device or a simulator configured to use the device architectures we build for.

Cargo test in this case is unfortunately not enough to tell if this will work on the intended devices, the above will.

**Comment by StriderDM** [comment]
There is also an alternative to this since the purpose of it is purely for tests. Since the libraries are not being static linked in to the wallet_ffi library by building with cargo, having the dependencies installed beforehand and in the path will be sufficient for the tests to run for wallet_ffi.

Just another option.

**Review by philipr-za — APPROVED** [review]

**Pull request merged**
